### DISCUSSION TITLE
0.9.1 using external MariaDB/MySQL

### DISCUSSION THREAD

**Discussion by LairdTurner** [ckulka/baikal-docker#76] [Q&A] [answered]
Hi!

I've configured baikal-docker and it starts up.
Now I want to use my MariaDB installed on the server itself. So for the MYSQL_HOST I enter the ip for docker0 (bind-address).
All was setup correctly a db called baikal etc. But I can't get any connection.
After a while I get an 504 Gateway Time-out.

Is there maybe a way to define the db within the docker-compose file?

Thanks!

Best regards

**Comment by ckulka** [comment, marked as answer]
Hmm, based on https://stackoverflow.com/a/48547074 that IP address should work, but double check with the following commands if the IP address is the correct one. The following assumes that you use the default bridge network mode (see "DRIVER" column):

```sh
# Lists all networks, only the "bridge" ones are important
$ docker network ls
NETWORK ID     NAME               DRIVER    SCOPE
71ea53e8e769   bridge             bridge    local
9fda14c20e2b   baikal_default     bridge    local
268e9280dbb5   host               host      local
f348769d07z8   none               null      local
```

If you use Docker Compose, then use the network with the folder name (or the name you gave it), otherwise use `bridge`.

If the driver is not `bridge`, then you can change that in the `docker-compose.yaml` using [`network_mode: bridge`](https://docs.docker.com/compose/compose-file/compose-file-v3/#network_mode).

After that, you can verify which network the Baikal container uses with

```bash
$ docker inspect <container name or id>

$ docker network inspect <network name or id, e.g. baikal_default>
# Note the "Gateway", that's the IP that you should use
# (...)
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
# (...)
```

If there was just a small typo/wrong network and it now works, I'd recommend to use an alias instead of the hardcoded IP address. I had IP addresses change on me when recreating/restoring Docker networks, aliases make your life safer there. Easiest is adding `extra_hosts` to the Baikal container to map the IP address to a name, see <https://docs.docker.com/compose/compose-file/compose-file-v2/#extra_hosts>.

If the IP address and port numbers are correct, then I don't have another idea as to why this wouldn't work other than local firewall rules.

I then only see two workarounds to get it to work:

1. Use the `host` network mode, so that you can use `127.0.0.1`/`localhost` as hostname (which I wouldn't recommend for a couple of reasons, so I'm not going into details unless you need/want to)
2. Add a MySQL container in the Docker Compose file

There's an example file for option two here: <https://github.com/ckulka/baikal-docker/blob/master/examples/docker-compose.mariadb.yaml>. It uses MariaDB, but you can change the image (<https://hub.docker.com/_/mysql>) and you should be good.

**Reply by ckulka** [reply]
Marked as the answer based on <https://github.com/ckulka/baikal-docker/discussions/76#discussioncomment-2188283>

**Comment by LairdTurner** [comment]
Hi!

If I enter docker network inspect baikal_default
then I get...
                {
                    "Subnet": "192.168.96.0/20",
                    "Gateway": "192.168.96.1"
                }
 
That's not what I expected...
I expected...
                {
                    "Subnet": "172.17.0.0/16",
                    "Gateway": "172.17.0.1"
                }

I like not use a containered db. I whant to use the locally installed db on the server.
What I meant was to configure the internal db from the compose file, so the initial setup don't ask me
to use SQLite or MySQL/MariaDB. Or is this generally not possible?

Thanks!

**Comment by LairdTurner** [comment]
Hi!

I added "network_mode: bridge" to the compose file and now I can reach the MariaDB!

**Reply by ckulka** [reply]
Top, glad to hear it worked out 👍
